refactor(Inventory Purchasable)

lomamech opened the pull request:
- Implement the way to filter inventory who requires a Purchase Order
- Update API for stock core filter the inventorys who Quantity is sup to
  zero when require_po is true
- Update E2E test for inventory stock add test for the Inventories who
  require purchase order and use property resetFilters for improve some
  test

closes #2251

lomamech committed:
refactor(Inventory Purchasable)
- Implement the way to filter inventory who requires a Purchase Order
- Update API for stock core filter the inventorys who Quantity is sup to
  zero when require_po is true
- Update E2E test for inventory stock add test for the Inventories who
  require purchase order and use property resetFilters for improve some
  test

jniles commented on server/controllers/stock/core.js:
```diff
@@ -450,6 +456,11 @@ function getInventoryQuantityAndConsumption(params) {
         return rows.filter(row => row.status === status);
       }
+
+      if(require_po) {
```
It looks like `status and `require_po` are exclusive.  Can both options be set?  If so, these shouldn't `return rows.filter()`, they should do something like:
```js
var filteredRows = rows;

if (status) {
  filteredRows = filteredRows.filter(row => row.status === status);
}

if (require_po) {
  filteredRows = filteredRows.filter(row => row.S_Q > 0);
}

return filteredRows;
```

jniles commented on client/src/modules/stock/inventories/modals/search.modal.html:
```diff
@@ -31,6 +31,15 @@
             <bh-clear on-clear="$ctrl.clear('inventory_uuid')"></bh-clear>
           </bh-inventory-select>
 
+          <div class="form-group checkbox">
```
There's no need for the `form-group` class here.

jniles reviewed (commented):
@lomamech, this code is pretty good!

I'm curious - shouldn't almost all the inventory be needing purchase as soon as one stock item is consumed?  It looks like this basically filters on `quantity != S_MAX`.

If so, is it really that useful to have a clientside filter?  The server-side filter is useful because we can then populate a PO for a client, but I'm not so sure the checkbox is useful.  What do you think?

lomamech committed:
refactor(Inventory Purchasable)
- Sanitaze code remove not necessary form-group
- refactor the filtering of purchases status and inventories who need PO

jniles reviewed (approved):
LGTM!

bors r+
